custom_components/ha-glocaltokens/api.py

```diff
@@ -46,3 +54,66 @@ def get_google_devices_json(self):
     def get_android_id(self):
         """Generate random android_id"""
         return self._client._get_android_id()
+
+    def format_timer_information(self, timer):
+        timestamp = convert_from_ms_to_s(timer[FIRE_TIME])
+        timer[DATE_TIME] = datetime.fromtimestamp(timestamp).strftime(SHOW_TIME_ONLY)
+
+        duration = convert_from_ms_to_s(timer[ORIGINAL_DURATION])
+        timer[DURATION] = datetime.utcfromtimestamp(duration).strftime(SHOW_TIME_ONLY)
+        return timer
+
+    def format_alarm_information(self, alarm):
+        timestamp = convert_from_ms_to_s(alarm[FIRE_TIME])
+        alarm[DATE_TIME] = datetime.utcfromtimestamp(timestamp).strftime(SHOW_DATE_TIMEZONE)
+        alarm[LOCAL_TIME] = datetime.fromtimestamp(timestamp).strftime(SHOW_DATE_AND_TIME)
+        return alarm
+
+    def create_url(self, ip, api_endpoint):
+        url = 'https://{ip}:{port}/{endpoint}'.format(
+            ip=ip,
+            port=str(PORT),
+            endpoint=api_endpoint
+        )
+        return url
+
+    def get_alarms_and_timers_from(self, ip, token, endpoint):
+
+        url = self.create_url(ip, endpoint)
+
+        HEADERS[HEADER_CAST_LOCAL_AUTH] = token
+
+        response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT)
+
+        return response
+
+    def get_google_devices_information(self):
+
+        devices = self._client.get_google_devices_json()
+
+        for device in devices:
+            local_token = device[TOKEN]
+
+            device[TIMERS] = []
+            device[ALARMS] = []
+
+            response = self.get_alarms_and_timers_from('192.168.0.205', local_token, API_ENDPOINT_ALARMS) # IP
+
+            if response.status_code != HTTP_OK:
+                _LOGGER.error("For device {device} - API returned {error}".format(device=device[DEVICE_NAME], error=response.status_code))
+                continue
+
+            result = response.json()
+
+            if TIMERS not in result and ALARMS not in result:
+                _LOGGER.error("For device {device} - {error}".format(device=device[DEVICE_NAME], error=API_RETURNED_UNKNOWN))
+                continue
+
+            timers = result[TIMERS]
+            device[TIMERS] = [self.format_timer_information(timer) for timer in timers]
+
+            alarms = result[ALARMS]
+            device[ALARMS] = [self.format_alarm_information(alarm) for alarm in alarms]
```

Comment:
To be honest I think we need to move `format_alarm_information` into `sensors` file. Now we seem to repeat ourselves. We format the timestamp here, in `api` and then in sensors we call `self._state = convert_from_ms_to_s(self._alarms[0][FIRE_TIME])` again that does the same conversion/formatting. Unless I am missing something.

I propose we keep `api` very blunt. Like:
- we make a request
- we receive data
- we store data (as we have received it) 

And in `sensors` file, we take that received data (as it is) and format it in the way we want it to be displayed/used around Home Assistant.
That way we decouple APIs from our logic. I think it would be a bit of a better approach, because it will allow us to go back into `sensors` any time and change the formating or manipulate received data as we want, without the need to look at `api` file, because there we would simply receive data as is and store it as is, without changing it.

Reply:
The reason i did it that way was to preserve the original attributes from google home, yes its been called twice, but different things are done each time.

I think your are right about making the API only fetch data and not processing it. I have fixed it now